### ISSUE TITLE
GUI doesn't display collections within orgs

### ISSUE THREAD

**Issue by predictiple** [Velocidex/velociraptor#2084]
version: 0.6.7-dev
commit: 38c77ec8
build_time: "2022-09-13T17:40:59Z"
compiler: go1.19.1

With server and client both the above version there seems to be a GUI bug as can be seen in the following screen capture. This issue only occurs with non-root orgs - the same client when associated with the root org works correctly.

https://user-images.githubusercontent.com/9843422/190011368-bf1c8413-6f1f-4278-9ff0-00ce7b9f0872.mp4

The collections are running and completing successfully, i.e. the data is present in the datastore within the org folder structure, so it seems to just be a display issue / GUI bug.

The behaviour when running collections (including the initial Generic.Client.Info) is that the flows either don't appear at all in the list or they appear very briefly before disappearing. The screen capture is of one instance where the flow briefly appeared and I managed to click on it. As can be seen the flow details are then shown in the lower pane and persist there but the flow then disappears from the upper list pane.

**Comment by scudette** [comment]
It looks like the GUI is switching orgs. Do you have another tab open? Do you have anything that might interfere with passing the org header? Can you verify that the user option is kept at the correct org after it switched?

Are you running on the read only data store?

**Comment by predictiple** [comment]
I definitely only have 1 tab open. The org header (`grpc-metadata-orgid` I assume?) seems to be present in all requests and I'm connecting from my home machine to a cloud VM so there shouldn't be any content interference along the way.

I've tried as much as possible to simplify everything so the setup for the testing is as follows:
- 1 org created via initial_org, plus the root org obviously. The server is using MemcacheFileDataStore and the folder structure on disk is cleared completely before starting the velo service.
- 1 client running on the server itself; same binary as the server. To switch "clients" I delete the writeback and swap the client.config.yaml. There are no issues with the clients connecting and they show up in the correct org, so the config is working. The client does receive the flow tasking (s seen in the -v logging) and doesn't report any errors.

Once the server was running I did this: switched to the new org and went to Server Artifacts. Created a collection of the Server.Orgs.ListOrgs artifact and selected the parameter to download the client configs. Result: The flow doesn't show at all in the list of collections.

I then switched back to the root org and discovered that the flow is listed under the root org's Server Artifacts. The rows are zero and there are no uploaded files. The Logs panel does not show the log.

![F CCGNI6GPE8C50_1](https://user-images.githubusercontent.com/9843422/190091033-d2971851-be46-4578-894c-6701e31bfe6b.png)

In disk this is what I found for that FlowId:

![F CCGNI6GPE8C50](https://user-images.githubusercontent.com/9843422/190091382-7cde79f1-14a0-4c32-a523-a6ba6805a60c.png)

For comparison, if I run the Server.Orgs.ListOrgs artifact in the root org this is how the resulting files appear:

![F CCGNMHT9DI2SU](https://user-images.githubusercontent.com/9843422/190091908-d5e955e5-0be2-4b70-8418-91d44633da7a.png)

So the issue seems to apply to server flows too, when they are run from a non-root org.

Going back to the client, I connected it to the org and ran a simple artifact (LinuxSys.Users). The flow did not show up at all in the GUI but I grabbed the FlowId from the client's log and searched for it on disk. This is what I found:

![F CCGNOU84VBGCQ](https://user-images.githubusercontent.com/9843422/190093002-7d1ace26-75d0-4bfc-8d9f-f792ed081dc0.png)

Although I previously though this was a GUI bug it now seems that maybe it isn't. The files being written outside the org's dir structure is probably throwing the GUI off.

It seems simple to reproduce, but let me know if there's anything else I can check?

**Comment by scudette** [comment]
This seems to be specifically an issue with the MemcacheFileDataStore - it does not happen with the regular FileBaseDataStore

**Comment by predictiple** [comment]
Yeah it does seem to be that. I've tried it now with FileBaseDataStore and the weird behaviour doesn't occur. I'll stick with that for further testing regarding the org stuff since the memcache is a frequent confounding factor.

This might be a separate issue: With FileBaseDataStore there is still some unexpected behaviour with Server.Orgs.ListOrgs when running it in an org, but at least now the flow is visible where it should be:
![F CCGP5L9802UQ8](https://user-images.githubusercontent.com/9843422/190108123-0f2eec90-435f-43d5-ad70-2fdc7fe591c0.png)

**Comment by scudette** [comment]
This behavior is expected - When you run a server artifact inside the org you take on the ACL of your user in the org. By default an admin user in an org does not have org admin permissions  so can not see or interact with any of the other orgs.

An admin user in the root org does receive the org admin permissions so they can manage orgs.

**Comment by scudette** [comment]
In any case thanks for the thorough testing! I submitted #2087 which should fix the issue on the memcache datastore.

**Comment by predictiple** [comment]
I've re-tested it with the fix (using Memcache datastore) and it resolves the issue. Thanks for the speedy response! :partying_face:
